### PR TITLE
Allow migrated Payment Methods to be charged on subscription renewal

### PR DESCRIPTION
**Pull request by dechov**
Fixes (edit: _not entirely_) https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1102
Fixes https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1181

#### Changes proposed in this Pull Request:

Allows `pm_` payment method IDs to be submitted as subscription payment method source (on Edit Subscription screen) and charged for renewals. (Other behaviors depending on `src_` IDs are unchanged.)

Adapted from solution offered in the comments of https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1181 (thanks @nsandrew).

To test:
- With WooCommerce Subscriptions activated…
- Create Customer with attached Payment Method, e.g. with cURL:
```bash
curl https://api.stripe.com/v1/payment_methods \
  -u sk_test_****: \
  -d type=card \
  -d "card[number]"=4242424242424242 \
  -d "card[exp_month]"=02 \
  -d "card[exp_year]"=2042 \
  -d "card[cvc]"=424
curl https://api.stripe.com/v1/customers \
  -u sk_test_****: \
  -d payment_method="pm_****"
```
- Navigate to the Edit Subscription screen for a new or existing active subscription
- Edit the Billing section, and change "Stripe Customer ID" and "Stripe Source ID" to the Customer and Payment Method IDs (respectively) created above
- Select "Process renewal" action, Update, and Verify that the payment went through

-------------------
- [ ] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

### CODE REVIEW

**Review by RadoslavGeorgiev — APPROVED** [review]
I tested this on my local setup and the PR seems to enable the usage of existing subscriptions with payment methods. Good job! :shipit: 

However, https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1102 requires some more attention before it can be closed. Please check it for further details.

**Comment by johnreboot** [comment]
Hi. We've got a large subscription migration project from Paid Memberships Pro to WooCommerce on hold until this plugin update is available. Any idea when that might happen? Thanks

**Comment by dechov** [comment]
Hi @johnreboot – this change is slated for a release very early next week, if not today.

**Comment by johnreboot** [comment]
That's great news! Thank you

**Comment by paultannenbaum** [comment]
@dechov Any update on when this will be released? Like @johnreboot this is a blocker for a migration project I am working on. If not sure, is there any docs on how we can compile `master` branch and run the build ourselves?

**Comment by onurk7** [comment]
We are also waiting for it and in the next 2 weeks, we have to give a decision. And we are afraid of crashing or doing something wrong on codes and potential issues may happen after the update of the plugin. It's an urgent issue for us and waiting for it almost one year to move to woocommerce subscriptions plugin.

**Comment by dechov** [comment]
Apologies for the delay – the change has gone out in the latest release (yesterday). Thank you for following up @johnreboot @paultannenbaum @psaikali @onurk7

**Comment by paultannenbaum** [comment]
@onurk7 They just issued a release yesterday evening. I am currently in the process of testing it out to see if it works for saving the payment method id (i.e. `pm_xxx`) under the "Stripe Source ID" field in the subscription billing field. I will report back once I have verified for our use case.

**Comment by paultannenbaum** [comment]
@dechov Thank you guys for putting out the release!

**Comment by onurk7** [comment]
Oh! Good news! We can prepare ourselves too then. @paultannenbaum can you please share your experience after your test? Thanks for the merge and let us know @dechov

**Comment by paultannenbaum** [comment]
@onurk7 I can confirm after testing both the subscriptions importer as well as editing existing subscription methods, that this works for saving payment method ids under the stripe source id field. We tested several different types of payment methods, and all were successfully charged using the updated plugin.

**Comment by onurk7** [comment]
Thanks @paultannenbaum !

**Comment by oo-matty** [comment]
@paultannenbaum did you use a commercial plug-in to import the subscriptions or something homebrew?

**Comment by paultannenbaum** [comment]
@oo-matty We used the [woocommerce-subscriptions-importer-exporter](https://github.com/woocommerce/woocommerce-subscriptions-importer-exporter) plugin, which worked very nicely for our needs.

**Comment by fsekerli** [comment]
Hi @paultannenbaum ! We used the same plugin to import Bold Subscriptions (from Shopify) to WooCommerce Subscriptions. Now, we have only customer id (cus_) and when we import everything and activate Stipe payments, in the test mode, if we proceed the renewal, there is an error. 

Recurring payment for order failed. Reason: You passed an empty string for 'payment_method'. We assume empty values are an attempt to unset a parameter; however 'payment_method' cannot be unset. You should remove 'payment_method' from your request or supply a non-empty value.

Any idea what to do here? Thanks!

**Pull request merged**
